core/src/main/scala/kafka/server/OffsetCheckpoint.scala

```diff
@@ -50,14 +52,20 @@ class OffsetCheckpoint(val file: File) extends Logging {
           writer.write("%s %d %d".format(topicPart.topic, topicPart.partition, offset))
           writer.newLine()
         }
-      
+
         // flush the buffer and then fsync the underlying file
         writer.flush()
         fileOutputStream.getFD().sync()
+      } catch {
+        case e: FileNotFoundException =>
+          if (FileSystems.getDefault.isReadOnly) {
+            fatal("Halting writes to offset checkpoint file because the underlying file system is inaccessible : ", e)
+            Runtime.getRuntime.halt(1)
+          }
```

Comment:
I think this is a bigger issue how we do halting upon errors. @MayureshGharat could you file a separate jira?